### PR TITLE
adding restriction to the request

### LINKED ISSUE #6: Add role support and restrict Food access
Only allow ADMIN role to create, edit or delete Foods

### PR DESCRIPTION
**Pull request by anaCode26**
closes #6

### CODE REVIEW

File: pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532.xml [hunk 1]
```diff
@@ -3,9 +3,17 @@
   <database-model serializer="dbm" dbms="POSTGRES" family-id="POSTGRES" format-version="4.53">
```
Comment (rdiazutn) [inline pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532.xml:3]: Should this file be commited?

File: pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532/storage_v2/_src_/database/postgres.edMnLQ/schema/public.abK9xQ.meta [hunk 2]
```diff
@@ -1,2 +1,2 @@
 #n:public
-!<md> [796, 0, null, null, -2147483648, -2147483648]
```
Comment (rdiazutn) [inline pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532/storage_v2/_src_/database/postgres.edMnLQ/schema/public.abK9xQ.meta:2]: Same question

File: pets-project/restservice/src/main/java/com/example/restservice/pet/controller/PetController.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import com.example.restservice.service.OwnerService;
 import com.example.restservice.service.PetService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
```
Comment (rdiazutn) [inline pets-project/restservice/src/main/java/com/example/restservice/pet/controller/PetController.java:10]: Is this import necessary?

File: pets-project/restservice/src/main/java/com/example/restservice/pet/model/Pet.java [hunk 4]
```diff
@@ -3,7 +3,13 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
 
+@Setter
+@Getter
+@AllArgsConstructor
```
Comment (rdiazutn) [inline pets-project/restservice/src/main/java/com/example/restservice/pet/model/Pet.java:12]: Why did you remove all the getters to do this? isn't it inconsistent with the rest of the tables?

File: pets-project/restservice/src/main/java/com/example/restservice/security/dto/LoginDto.java [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 public record LoginDto (
         String email,
-        String password
+        String password,
+        String role
```
Comment (rdiazutn) [inline pets-project/restservice/src/main/java/com/example/restservice/security/dto/LoginDto.java:6]: Why did you add this change?

**Review by rdiazutn — CHANGES_REQUESTED** [review]
Leaving some comments

File: pets-project/restservice/src/main/java/com/example/restservice/pet/controller/PetController.java (hunk 3 above)
Comment (anaCode26) [reply pets-project/restservice/src/main/java/com/example/restservice/pet/controller/PetController.java:10]: Deleted!

File: pets-project/restservice/src/main/java/com/example/restservice/pet/model/Pet.java (hunk 4 above)
Comment (anaCode26) [reply pets-project/restservice/src/main/java/com/example/restservice/pet/model/Pet.java:12]: Replaced the Getter and Setter annotations for methods.

File: pets-project/restservice/src/main/java/com/example/restservice/security/dto/LoginDto.java (hunk 5 above)
Comment (anaCode26) [reply pets-project/restservice/src/main/java/com/example/restservice/security/dto/LoginDto.java:6]: I can't remember xd

File: pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532.xml (hunk 1 above)
Comment (anaCode26) [reply pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532.xml:3]: mmm I forgot to ignore it.

File: pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532/storage_v2/_src_/database/postgres.edMnLQ/schema/public.abK9xQ.meta (hunk 2 above)
Comment (anaCode26) [reply pets-project/.idea/dataSources/d91c7433-9d85-44df-9f99-a12e10c9e532/storage_v2/_src_/database/postgres.edMnLQ/schema/public.abK9xQ.meta:2]: Same answer

**Review by rdiazutn — APPROVED** [review]
